packages/sbtc/src/api.ts

```diff
@@ -0,0 +1,359 @@
+      bitcoinTxid: string;
+      bitcoinTxOutputIndex: number;
+      recipient: string;
+      amount: number;
+      lastUpdateHeight: number;
+      lastUpdateBlockHash: string;
+      status: string;
+      statusMessage: string;
+      parameters: {
+        maxFee: number;
+        lockTime: number;
+      };
+      reclaimScript: string;
+      depositScript: string;
+    };
+  }
+
+  async fetchSignersPublicKey(contractAddress?: string): Promise<string> {
+    const res = (await fetchCallReadOnlyFunction({
+      contractAddress: contractAddress ?? this.config.sbtcContract,
+      contractName: 'sbtc-registry',
+      functionName: 'get-current-aggregate-pubkey',
+      functionArgs: [],
+      senderAddress: STACKS_DEVNET.bootAddress, // zero address
+      client: {
+        baseUrl: this.config.stxApiUrl,
+      },
+    })) as BufferCV;
+
+    return res.value.slice(2);
+  }
+
+  async fetchSignersAddress(contractAddress?: string): Promise<string> {
+    const pub = await this.fetchSignersPublicKey(contractAddress ?? this.config.sbtcContract);
+    return btc.p2tr(pub, undefined, REGTEST).address!;
+  }
+
+  async fetchCallReadOnly({
+    contractAddress,
+    functionName,
+    args = [],
+    sender = STACKS_DEVNET.bootAddress, // zero address
+  }: {
+    contractAddress: string;
+    functionName: string;
+    args?: ClarityValue[];
+    sender?: string;
+  }) {
+    contractAddress = contractAddress.replace('.', '/');
+    return await fetch(
+      `${this.config.stxApiUrl}/v2/contracts/call-read/${contractAddress}/${encodeURIComponent(
+        functionName
+      )}`,
+      {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ sender, arguments: args.map(Cl.serialize) }),
+      }
+    )
+      .then(res => res.json())
+      .then(res => Cl.deserialize(res.result));
+  }
+
+  /** Get BTC balance (in satoshis) */
+  async fetchBalance(address: string): Promise<number> {
+    // todo: check if better endpoints now exist
+    const addressInfo = await fetch(`${this.config.btcApiUrl}/address/${address}`).then(r =>
+      r.json()
+    );
+
+    return addressInfo.chain_stats.funded_txo_sum - addressInfo.chain_stats.spent_txo_sum;
+  }
+
+  async fetchSbtcBalance(stacksAddress: string) {
+    const balance = (await this.fetchCallReadOnly({
+      contractAddress: this.config.sbtcContract,
+      functionName: 'get-balance',
+      args: [Cl.address(stacksAddress)],
+    })) as SomeCV<UIntCV>;
+
+    return balance?.value?.value ?? 0;
+  }
+}
+
+/** todo */
+export class SbtcApiClientTestnet extends SbtcApiClient {
+  constructor(config?: Partial<BaseClientConfig>) {
+    super(
+      Object.assign(
+        {
+          sbtcApiUrl: 'https://beta.sbtc-emily.com', // todo: get real url
+          btcApiUrl: 'https://TODO', // todo: replace with functioning regtest testnet deployment
```

Comment:
@janniks I've gotten pretty far in that I've built the deposit tx in our wallet and can successfully sign/finalize/broadcast it. I'm using `https://beta.sbtc-mempool.tech/api/proxy` here, is that correct, bc I do get a txid back from `broadcastTx`, but then I get an error from the `sbtcApiUrl` calling the notify function. I just wanted to confirm, this all expected on testnet atm?

Reply:
Try the latest version and snippets from npm. I think that should work, but I'll double check. I did change something on the sbtc emily deposit notify yesterday

Reply:
`https://staging.beta.sbtc.tech/api/proxy` is the one I was told about

Reply:
It should actually be beta.sbtc-mempool.tech. They currently all point behind the same host, but that is not guaranteed.

Reply:
Using...
```ts
const client = new SbtcApiClientTestnet({
  sbtcApiUrl: 'https://beta.sbtc-mempool.tech/api/proxy',
  btcApiUrl: 'https://beta.sbtc-mempool.tech/api/proxy',
  stxApiUrl: 'https://api.testnet.hiro.so',
  sbtcContract: 'SNGWPN3XDAQE673MXYXF81016M50NHF5X5PWWM70',
});
```
I get a 500 error on the notify...
<img width="666" alt="Screenshot 2024-12-04 at 8 03 42 PM" src="https://github.com/user-attachments/assets/0ac29e4d-7595-4c1a-8890-389f70162d5a">

<img width="959" alt="Screenshot 2024-12-04 at 8 04 55 PM" src="https://github.com/user-attachments/assets/dd4e7c8b-f393-4f6f-bad8-6803bb3a861a">

cc @kyranjamie just to keep you in the loop here.

Reply:
I think this is a cors issue from the backend for the bridge, pushing a fix to update for needed setting on cors

Reply:
Reported in Slack, I'm still hitting errors. I tried with these two request urls...

![Screenshot 2024-12-06 at 8 58 50 AM](https://github.com/user-attachments/assets/d0a7383c-f717-4963-8dbe-265fe63d442b)

![Screenshot 2024-12-06 at 9 05 38 AM](https://github.com/user-attachments/assets/ded0152c-57c8-4a08-b15c-e4d0ec123479)

What am I missing?

Reply:
Apparently a status param needs to be included.